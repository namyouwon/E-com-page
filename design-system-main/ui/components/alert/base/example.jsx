// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import Alert from '../';
import { UtilityIcon } from '../../icons/base/example';
import EmptyLink from '../../../shared/empty-link/';
import Heading from '../../../shared/heading/';

export default [
  {
    id: 'default',
    label: 'Default',
    element: (
      <Alert>
        <UtilityIcon
          containerClassName="slds-m-right_x-small"
          className="slds-icon_x-small"
          assistiveText={false}
          symbol="user"
        />
        <Heading level="2">
          Logged in as John Smith (johnsmith@acme.com).{' '}
          <EmptyLink>Log out</EmptyLink>
        </Heading>
      </Alert>
    )
  }
];

export let states = [
  {
    id: 'theme-info',
    label: 'Default (Theme Class  - Deprecated)',
    element: (
      <Alert isLegacy>
        <UtilityIcon
          containerClassName="slds-m-right_x-small"
          className="slds-icon_x-small"
          assistiveText={false}
          symbol="user"
        />
        <Heading level="2">
          Logged in as John Smith (johnsmith@acme.com).{' '}
          <EmptyLink>Log out</EmptyLink>
        </Heading>
      </Alert>
    )
  },
  {
    id: 'warning',
    label: 'Warning',
    element: (
      <Alert type="warning">
        <UtilityIcon
          containerClassName="slds-m-right_x-small"
          className="slds-icon_x-small"
          assistiveText={false}
          symbol="warning"
        />
        <Heading level="2">
          Your browser is outdated. Your Salesforce experience may be degraded.{' '}
          <EmptyLink>More Information</EmptyLink>
        </Heading>
      </Alert>
    )
  },
  {
    id: 'theme-warning',
    label: 'Warning (Theme Class  - Deprecated)',
    element: (
      <Alert type="warning" isLegacy>
        <UtilityIcon
          containerClassName="slds-m-right_x-small"
          className="slds-icon_x-small"
          assistiveText={false}
          symbol="warning"
        />
        <Heading level="2">
          Your browser is outdated. Your Salesforce experience may be degraded.{' '}
          <EmptyLink>More Information</EmptyLink>
        </Heading>
      </Alert>
    )
  },
  {
    id: 'error',
    label: 'Error',
    element: (
      <Alert type="error">
        <UtilityIcon
          containerClassName="slds-m-right_x-small"
          className="slds-icon_x-small"
          assistiveText={false}
          symbol="error"
        />
        <Heading level="2">
          Your browser is currently not supported. Your Salesforce may be
          degraded. <EmptyLink>More Information</EmptyLink>
        </Heading>
      </Alert>
    )
  },
  {
    id: 'theme-error',
    label: 'Error (Theme Class  - Deprecated)',
    element: (
      <Alert type="error" isLegacy>
        <UtilityIcon
          containerClassName="slds-m-right_x-small"
          className="slds-icon_x-small"
          assistiveText={false}
          symbol="error"
        />
        <Heading level="2">
          Your browser is currently not supported. Your Salesforce may be
          degraded. <EmptyLink>More Information</EmptyLink>
        </Heading>
      </Alert>
    )
  },
  {
    id: 'offline',
    label: 'Offline',
    element: (
      <Alert type="offline">
        <UtilityIcon
          containerClassName="slds-m-right_x-small"
          className="slds-icon_x-small"
          assistiveText={false}
          symbol="offline"
        />
        <Heading level="2">
          You are in offline mode. <EmptyLink>More Information</EmptyLink>
        </Heading>
      </Alert>
    )
  },
  {
    id: 'theme-offline',
    label: 'Offline (Theme Class  - Deprecated)',
    element: (
      <Alert type="offline" isLegacy>
        <UtilityIcon
          containerClassName="slds-m-right_x-small"
          className="slds-icon_x-small"
          assistiveText={false}
          symbol="offline"
        />
        <Heading level="2">
          You are in offline mode. <EmptyLink>More Information</EmptyLink>
        </Heading>
      </Alert>
    )
  }
];
