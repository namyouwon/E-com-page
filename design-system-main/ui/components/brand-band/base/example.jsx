// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';

import BrandBand from '../';

export default <BrandBand size="medium" />;

export let examples = [
  {
    id: 'brand-band-small',
    label: 'Small',
    element: <BrandBand size="small" />
  },
  {
    id: 'brand-band-large',
    label: 'Large',
    element: <BrandBand size="large" />
  },
  {
    id: 'brand-band-no-image',
    label: 'No Image',
    element: <BrandBand size="none" />
  },
  {
    id: 'brand-band-large-group',
    label: 'Large Group',
    element: <BrandBand size="large" image="group" />
  },
  {
    id: 'brand-band-large-user',
    label: 'Large User',
    element: <BrandBand size="large" image="user" />
  }
];
