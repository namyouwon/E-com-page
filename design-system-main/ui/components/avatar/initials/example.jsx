// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import { Avatar } from '../base/example';
import { AvatarInitials } from '../';

export default [
  {
    id: 'default',
    label: 'Default',
    element: (
      <Avatar>
        <AvatarInitials
          iconClassName="slds-icon-standard-account"
          title="company name"
          initials="Ac"
        />
      </Avatar>
    )
  }
];

export let examples = [
  {
    id: 'linked-avatar-initials',
    label: 'Wrapped in a link',
    element: (
      <a href="#" onClick={e => e.preventDefault()}>
        <Avatar>
          <AvatarInitials
            iconClassName="slds-icon-standard-account"
            title="company name"
            initials="Ac"
          />
        </Avatar>
      </a>
    )
  }
];
