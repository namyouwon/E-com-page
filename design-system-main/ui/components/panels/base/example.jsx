// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import Panel, { PanelPlayground, HeaderDeprecated } from '../';
import { ActionOverflow } from '../../menus/dropdown/example';

export default <PanelPlayground />;

// overflow: hidden hides bleeding box-shadow (demo only)
const sharedDemoStyles = `
  background-color: #fafaf9;
  overflow: hidden;
`;

export let examples = [
  {
    id: 'with-centered-title',
    label: 'With Centered Title',
    element: <PanelPlayground hasCenterTitle />
  },
  {
    id: 'with-truncated-title',
    label: 'With Truncated Title',
    element: (
      <PanelPlayground title="Panel Header with a really, really long name" />
    )
  },
  {
    id: 'with-secondary-actions',
    label: 'With Secondary Actions',
    demoStyles: `${sharedDemoStyles} height: 240px;`,
    storybookStyles: true,
    element: (
      <Panel
        size="medium"
        title="Panel Header"
        docked="left"
        invoke="toggle"
        headerActions={<ActionOverflow position="right" size="small" isOpen />}
      >
        A panel body accepts any layout or component
      </Panel>
    )
  },
  {
    id: 'with-drilled-in-left',
    label: 'With Drilled-In State - Position Left',
    demoStyles: `${sharedDemoStyles} height: 200px;`,
    storybookStyles: true,
    element: (
      <Panel size="medium" title="Panel Header" docked="left" invoke="drill-in">
        A panel body accepts any layout or component
      </Panel>
    )
  },
  {
    id: 'with-drilled-in-left-secondary-actions',
    label: 'With Drilled-In State & Secondary Actions - Position left',
    demoStyles: `${sharedDemoStyles} height: 200px;`,
    storybookStyles: true,
    element: (
      <Panel
        size="medium"
        title="Panel Header"
        docked="left"
        invoke="drill-in"
        headerActions={<ActionOverflow size="small" position="right" isOpen />}
      >
        A panel body accepts any layout or component
      </Panel>
    )
  },
  {
    id: 'with-drilled-in-right',
    label: 'With Drilled-In State - Position Right',
    demoStyles: `${sharedDemoStyles} height: 200px;`,
    storybookStyles: true,
    element: (
      <Panel
        size="medium"
        title="Panel Header"
        docked="right"
        invoke="drill-in"
      >
        A panel body accepts any layout or component
      </Panel>
    )
  },
  {
    id: 'drawer',
    label: 'Open as a drawer',
    element: <PanelPlayground drawer />
  },
  {
    id: 'deprecated-panel-header',
    label: 'DEPRECATED - Panel Header',
    element: (
      <Panel size="medium" docked="left" headerSlot={<HeaderDeprecated />}>
        A panel body accepts any layout or component
      </Panel>
    )
  }
];
