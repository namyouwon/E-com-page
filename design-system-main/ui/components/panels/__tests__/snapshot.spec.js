/* eslint-env jest */
import React from 'react';
import createHelpers from '../../../../jest.helpers';
import Panel from '../';
import { getDisplayElementById } from '../../../shared/helpers';
import { examples } from '../base/example';

const { matchesMarkup } = createHelpers(__dirname);
const testContainerStyles = {
  backgroundColor: '#fafaf9',
  position: 'relative',
  height: '600px',
  overflow: 'hidden'
};

describe('render Panel component', () => {
  it('render Base Panel', () =>
    matchesMarkup(
      <div style={testContainerStyles}>
        <Panel size="medium" title="Panel Header" docked="left" invoke="toggle">
          A panel body accepts any layout or component
        </Panel>
      </div>
    ));
  it('render Base Panel with left position drill in', () =>
    matchesMarkup(
      <div style={testContainerStyles}>
        <Panel
          size="medium"
          title="Panel Header"
          docked="left"
          invoke="drill-in"
        >
          A panel body accepts any layout or component
        </Panel>
      </div>
    ));
  it('render Base Panel with right position drill in', () =>
    matchesMarkup(
      <div style={testContainerStyles}>
        <Panel
          size="medium"
          title="Panel Header"
          docked="right"
          invoke="drill-in"
        >
          A panel body accepts any layout or component
        </Panel>
      </div>
    ));
  it('render Base Panel drawer', () =>
    matchesMarkup(
      <div style={testContainerStyles}>
        <Panel
          size="medium"
          title="Panel Header"
          docked="left"
          invoke="toggle"
          isStatic
        >
          A panel body accepts any layout or component
        </Panel>
      </div>
    ));
  it('render Base Panel small size', () =>
    matchesMarkup(
      <div style={testContainerStyles}>
        <Panel size="small" title="Panel Header" docked="left" invoke="toggle">
          A panel body accepts any layout or component
        </Panel>
      </div>
    ));
  it('render Base Panel medium size', () =>
    matchesMarkup(
      <div style={testContainerStyles}>
        <Panel size="medium" title="Panel Header" docked="left" invoke="toggle">
          A panel body accepts any layout or component
        </Panel>
      </div>
    ));
  it('render Base Panel large size', () =>
    matchesMarkup(
      <div style={testContainerStyles}>
        <Panel size="large" title="Panel Header" docked="left" invoke="toggle">
          A panel body accepts any layout or component
        </Panel>
      </div>
    ));
  it('render Base Panel x-large size', () =>
    matchesMarkup(
      <div style={testContainerStyles}>
        <Panel
          size="x-large"
          title="Panel Header"
          docked="left"
          invoke="toggle"
        >
          A panel body accepts any layout or component
        </Panel>
      </div>
    ));
  it('render Base Panel full size', () =>
    matchesMarkup(
      <div style={testContainerStyles}>
        <Panel size="full" title="Panel Header" docked="left" invoke="toggle">
          A panel body accepts any layout or component
        </Panel>
      </div>
    ));
  it('render Base Panel animated', () =>
    matchesMarkup(
      <div style={testContainerStyles}>
        <Panel isAnimated title="Panel Header" docked="left" invoke="toggle">
          A panel body accepts any layout or component
        </Panel>
      </div>
    ));
  it('renders Base Panel with centered title text', () =>
    matchesMarkup(
      <div style={testContainerStyles}>
        <Panel
          isAnimated
          title="Panel Header"
          docked="left"
          invoke="toggle"
          hasCenterTitle
        >
          A panel body accepts any layout or component
        </Panel>
      </div>
    ));
  it('renders Base Panel with long header title', () =>
    matchesMarkup(
      <div style={testContainerStyles}>
        <Panel
          isAnimated
          title="Panel Header with a really, really long title name"
          docked="left"
          invoke="toggle"
        >
          A panel body accepts any layout or component
        </Panel>
      </div>
    ));
  it('renders Base Panel with secondary actions', () =>
    matchesMarkup(getDisplayElementById(examples, 'with-secondary-actions')));
  it('renders Base Panel open as a drawer', () =>
    matchesMarkup(getDisplayElementById(examples, 'drawer')));
  it('renders Base Panel with the deprecated header', () =>
    matchesMarkup(getDisplayElementById(examples, 'deprecated-panel-header')));
});
