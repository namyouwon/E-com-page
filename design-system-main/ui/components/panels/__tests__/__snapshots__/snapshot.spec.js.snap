// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Panel component render Base Panel 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#fafaf9",
      "height": "600px",
      "overflow": "hidden",
      "position": "relative",
    }
  }
>
  <div
    aria-hidden={false}
    className="slds-panel slds-size_medium slds-panel_docked slds-panel_docked-left slds-is-open"
  >
    <div
      className="slds-panel__header"
    >
      <h2
        className="slds-panel__header-title slds-text-heading_small slds-truncate"
        title="Panel Header"
      >
        Panel Header
      </h2>
      <div
        className="slds-panel__header-actions"
      >
        <button
          className="slds-button slds-button_icon slds-button_icon-small slds-panel__close"
          title="Collapse Panel Header"
        >
          <svg
            aria-hidden={true}
            className="slds-button__icon"
          >
            <use
              xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#close"
            />
          </svg>
          <span
            className="slds-assistive-text"
          >
            Collapse Panel Header
          </span>
        </button>
      </div>
    </div>
    <div
      className="slds-panel__body"
    >
      A panel body accepts any layout or component
    </div>
  </div>
</div>
`;

exports[`render Panel component render Base Panel animated 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#fafaf9",
      "height": "600px",
      "overflow": "hidden",
      "position": "relative",
    }
  }
>
  <div
    aria-hidden={false}
    className="slds-panel slds-panel_docked slds-panel_docked-left slds-panel_animated slds-is-open"
  >
    <div
      className="slds-panel__header"
    >
      <h2
        className="slds-panel__header-title slds-text-heading_small slds-truncate"
        title="Panel Header"
      >
        Panel Header
      </h2>
      <div
        className="slds-panel__header-actions"
      >
        <button
          className="slds-button slds-button_icon slds-button_icon-small slds-panel__close"
          title="Collapse Panel Header"
        >
          <svg
            aria-hidden={true}
            className="slds-button__icon"
          >
            <use
              xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#close"
            />
          </svg>
          <span
            className="slds-assistive-text"
          >
            Collapse Panel Header
          </span>
        </button>
      </div>
    </div>
    <div
      className="slds-panel__body"
    >
      A panel body accepts any layout or component
    </div>
  </div>
</div>
`;

exports[`render Panel component render Base Panel drawer 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#fafaf9",
      "height": "600px",
      "overflow": "hidden",
      "position": "relative",
    }
  }
>
  <div
    aria-hidden={false}
    className="slds-panel slds-size_medium slds-panel_docked slds-panel_docked-left slds-is-open"
  >
    <div
      className="slds-panel__header"
    >
      <h2
        className="slds-panel__header-title slds-text-heading_small slds-truncate"
        title="Panel Header"
      >
        Panel Header
      </h2>
      <div
        className="slds-panel__header-actions"
      >
        <button
          className="slds-button slds-button_icon slds-button_icon-small slds-panel__close"
          title="Collapse Panel Header"
        >
          <svg
            aria-hidden={true}
            className="slds-button__icon"
          >
            <use
              xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#close"
            />
          </svg>
          <span
            className="slds-assistive-text"
          >
            Collapse Panel Header
          </span>
        </button>
      </div>
    </div>
    <div
      className="slds-panel__body"
    >
      A panel body accepts any layout or component
    </div>
  </div>
</div>
`;

exports[`render Panel component render Base Panel full size 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#fafaf9",
      "height": "600px",
      "overflow": "hidden",
      "position": "relative",
    }
  }
>
  <div
    aria-hidden={false}
    className="slds-panel slds-size_full slds-panel_docked slds-panel_docked-left slds-is-open"
  >
    <div
      className="slds-panel__header"
    >
      <h2
        className="slds-panel__header-title slds-text-heading_small slds-truncate"
        title="Panel Header"
      >
        Panel Header
      </h2>
      <div
        className="slds-panel__header-actions"
      >
        <button
          className="slds-button slds-button_icon slds-button_icon-small slds-panel__close"
          title="Collapse Panel Header"
        >
          <svg
            aria-hidden={true}
            className="slds-button__icon"
          >
            <use
              xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#close"
            />
          </svg>
          <span
            className="slds-assistive-text"
          >
            Collapse Panel Header
          </span>
        </button>
      </div>
    </div>
    <div
      className="slds-panel__body"
    >
      A panel body accepts any layout or component
    </div>
  </div>
</div>
`;

exports[`render Panel component render Base Panel large size 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#fafaf9",
      "height": "600px",
      "overflow": "hidden",
      "position": "relative",
    }
  }
>
  <div
    aria-hidden={false}
    className="slds-panel slds-size_large slds-panel_docked slds-panel_docked-left slds-is-open"
  >
    <div
      className="slds-panel__header"
    >
      <h2
        className="slds-panel__header-title slds-text-heading_small slds-truncate"
        title="Panel Header"
      >
        Panel Header
      </h2>
      <div
        className="slds-panel__header-actions"
      >
        <button
          className="slds-button slds-button_icon slds-button_icon-small slds-panel__close"
          title="Collapse Panel Header"
        >
          <svg
            aria-hidden={true}
            className="slds-button__icon"
          >
            <use
              xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#close"
            />
          </svg>
          <span
            className="slds-assistive-text"
          >
            Collapse Panel Header
          </span>
        </button>
      </div>
    </div>
    <div
      className="slds-panel__body"
    >
      A panel body accepts any layout or component
    </div>
  </div>
</div>
`;

exports[`render Panel component render Base Panel medium size 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#fafaf9",
      "height": "600px",
      "overflow": "hidden",
      "position": "relative",
    }
  }
>
  <div
    aria-hidden={false}
    className="slds-panel slds-size_medium slds-panel_docked slds-panel_docked-left slds-is-open"
  >
    <div
      className="slds-panel__header"
    >
      <h2
        className="slds-panel__header-title slds-text-heading_small slds-truncate"
        title="Panel Header"
      >
        Panel Header
      </h2>
      <div
        className="slds-panel__header-actions"
      >
        <button
          className="slds-button slds-button_icon slds-button_icon-small slds-panel__close"
          title="Collapse Panel Header"
        >
          <svg
            aria-hidden={true}
            className="slds-button__icon"
          >
            <use
              xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#close"
            />
          </svg>
          <span
            className="slds-assistive-text"
          >
            Collapse Panel Header
          </span>
        </button>
      </div>
    </div>
    <div
      className="slds-panel__body"
    >
      A panel body accepts any layout or component
    </div>
  </div>
</div>
`;

exports[`render Panel component render Base Panel small size 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#fafaf9",
      "height": "600px",
      "overflow": "hidden",
      "position": "relative",
    }
  }
>
  <div
    aria-hidden={false}
    className="slds-panel slds-size_small slds-panel_docked slds-panel_docked-left slds-is-open"
  >
    <div
      className="slds-panel__header"
    >
      <h2
        className="slds-panel__header-title slds-text-heading_small slds-truncate"
        title="Panel Header"
      >
        Panel Header
      </h2>
      <div
        className="slds-panel__header-actions"
      >
        <button
          className="slds-button slds-button_icon slds-button_icon-small slds-panel__close"
          title="Collapse Panel Header"
        >
          <svg
            aria-hidden={true}
            className="slds-button__icon"
          >
            <use
              xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#close"
            />
          </svg>
          <span
            className="slds-assistive-text"
          >
            Collapse Panel Header
          </span>
        </button>
      </div>
    </div>
    <div
      className="slds-panel__body"
    >
      A panel body accepts any layout or component
    </div>
  </div>
</div>
`;

exports[`render Panel component render Base Panel with left position drill in 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#fafaf9",
      "height": "600px",
      "overflow": "hidden",
      "position": "relative",
    }
  }
>
  <div
    aria-hidden={false}
    className="slds-panel slds-size_medium slds-panel_docked slds-panel_docked-left slds-is-open"
  >
    <div
      className="slds-panel__header"
    >
      <button
        className="slds-button slds-button_icon slds-button_icon-small slds-panel__back"
        title="Collapse Panel Header"
      >
        <svg
          aria-hidden={true}
          className="slds-button__icon"
        >
          <use
            xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#chevronleft"
          />
        </svg>
        <span
          className="slds-assistive-text"
        >
          Collapse Panel Header
        </span>
      </button>
      <h2
        className="slds-panel__header-title slds-text-heading_small slds-truncate"
        title="Panel Header"
      >
        Panel Header
      </h2>
      <div
        className="slds-panel__header-actions"
      >
        <button
          className="slds-button slds-button_icon slds-button_icon-small slds-panel__close"
          title="Collapse Panel Header"
        >
          <svg
            aria-hidden={true}
            className="slds-button__icon"
          >
            <use
              xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#close"
            />
          </svg>
          <span
            className="slds-assistive-text"
          >
            Collapse Panel Header
          </span>
        </button>
      </div>
    </div>
    <div
      className="slds-panel__body"
    >
      A panel body accepts any layout or component
    </div>
  </div>
</div>
`;

exports[`render Panel component render Base Panel with right position drill in 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#fafaf9",
      "height": "600px",
      "overflow": "hidden",
      "position": "relative",
    }
  }
>
  <div
    aria-hidden={false}
    className="slds-panel slds-size_medium slds-panel_docked slds-panel_docked-right slds-is-open"
  >
    <div
      className="slds-panel__header"
    >
      <button
        className="slds-button slds-button_icon slds-button_icon-small slds-panel__back"
        title="Collapse Panel Header"
      >
        <svg
          aria-hidden={true}
          className="slds-button__icon"
        >
          <use
            xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#chevronleft"
          />
        </svg>
        <span
          className="slds-assistive-text"
        >
          Collapse Panel Header
        </span>
      </button>
      <h2
        className="slds-panel__header-title slds-text-heading_small slds-truncate"
        title="Panel Header"
      >
        Panel Header
      </h2>
      <div
        className="slds-panel__header-actions"
      >
        <button
          className="slds-button slds-button_icon slds-button_icon-small slds-panel__close"
          title="Collapse Panel Header"
        >
          <svg
            aria-hidden={true}
            className="slds-button__icon"
          >
            <use
              xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#close"
            />
          </svg>
          <span
            className="slds-assistive-text"
          >
            Collapse Panel Header
          </span>
        </button>
      </div>
    </div>
    <div
      className="slds-panel__body"
    >
      A panel body accepts any layout or component
    </div>
  </div>
</div>
`;

exports[`render Panel component render Base Panel x-large size 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#fafaf9",
      "height": "600px",
      "overflow": "hidden",
      "position": "relative",
    }
  }
>
  <div
    aria-hidden={false}
    className="slds-panel slds-size_x-large slds-panel_docked slds-panel_docked-left slds-is-open"
  >
    <div
      className="slds-panel__header"
    >
      <h2
        className="slds-panel__header-title slds-text-heading_small slds-truncate"
        title="Panel Header"
      >
        Panel Header
      </h2>
      <div
        className="slds-panel__header-actions"
      >
        <button
          className="slds-button slds-button_icon slds-button_icon-small slds-panel__close"
          title="Collapse Panel Header"
        >
          <svg
            aria-hidden={true}
            className="slds-button__icon"
          >
            <use
              xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#close"
            />
          </svg>
          <span
            className="slds-assistive-text"
          >
            Collapse Panel Header
          </span>
        </button>
      </div>
    </div>
    <div
      className="slds-panel__body"
    >
      A panel body accepts any layout or component
    </div>
  </div>
</div>
`;

exports[`render Panel component renders Base Panel open as a drawer 1`] = `
Array [
  <header
    className="demo-only-element slds-theme_default slds-border_bottom slds-p-around_small"
    style={
      Object {
        "zIndex": "1",
      }
    }
  >
    <button
      aria-controls="example-unique-id-1"
      aria-expanded={true}
      aria-pressed={true}
      className="slds-button slds-button_icon slds-is-selected slds-button_icon-border-filled slds-button_icon-border"
      onClick={[Function]}
      title="Toggle panel"
    >
      <svg
        aria-hidden={true}
        className="slds-button__icon"
      >
        <use
          xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#toggle_panel_left"
        />
      </svg>
      <span
        className="slds-assistive-text"
      >
        Provide description of action
      </span>
    </button>
  </header>,
  <div
    className="demo-only-element"
    style={
      Object {
        "backgroundColor": "#fafaf9",
        "display": "flex",
        "flexDirection": false,
        "height": "600px",
        "overflow": "hidden",
        "position": "relative",
      }
    }
  >
    <div
      aria-hidden={false}
      className="slds-panel slds-size_medium slds-panel_docked slds-panel_docked-left slds-panel_drawer slds-is-open"
      id="example-unique-id-1"
    >
      <div
        className="slds-panel__header"
      >
        <h2
          className="slds-panel__header-title slds-text-heading_small slds-truncate"
          title="Panel Header"
        >
          Panel Header
        </h2>
        <div
          className="slds-panel__header-actions"
        >
          <button
            className="slds-button slds-button_icon slds-button_icon-small slds-panel__close"
            onClick={[Function]}
            title="Collapse Panel Header"
          >
            <svg
              aria-hidden={true}
              className="slds-button__icon"
            >
              <use
                xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#close"
              />
            </svg>
            <span
              className="slds-assistive-text"
            >
              Collapse Panel Header
            </span>
          </button>
        </div>
      </div>
      <div
        className="slds-panel__body"
      >
        A panel body accepts any layout or component
      </div>
    </div>
    <div
      className="demo-only-content slds-col slds-p-around_medium"
    >
      This section is demo-only content representing a custom layout in conjunction with the panel. It is not a part of the blueprint.
    </div>
  </div>,
]
`;

exports[`render Panel component renders Base Panel with centered title text 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#fafaf9",
      "height": "600px",
      "overflow": "hidden",
      "position": "relative",
    }
  }
>
  <div
    aria-hidden={false}
    className="slds-panel slds-panel_docked slds-panel_docked-left slds-panel_animated slds-is-open"
  >
    <div
      className="slds-panel__header slds-panel__header_align-center"
    >
      <h2
        className="slds-panel__header-title slds-text-heading_small slds-truncate"
        title="Panel Header"
      >
        Panel Header
      </h2>
      <div
        className="slds-panel__header-actions"
      >
        <button
          className="slds-button slds-button_icon slds-button_icon-small slds-panel__close"
          title="Collapse Panel Header"
        >
          <svg
            aria-hidden={true}
            className="slds-button__icon"
          >
            <use
              xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#close"
            />
          </svg>
          <span
            className="slds-assistive-text"
          >
            Collapse Panel Header
          </span>
        </button>
      </div>
    </div>
    <div
      className="slds-panel__body"
    >
      A panel body accepts any layout or component
    </div>
  </div>
</div>
`;

exports[`render Panel component renders Base Panel with long header title 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#fafaf9",
      "height": "600px",
      "overflow": "hidden",
      "position": "relative",
    }
  }
>
  <div
    aria-hidden={false}
    className="slds-panel slds-panel_docked slds-panel_docked-left slds-panel_animated slds-is-open"
  >
    <div
      className="slds-panel__header"
    >
      <h2
        className="slds-panel__header-title slds-text-heading_small slds-truncate"
        title="Panel Header with a really, really long title name"
      >
        Panel Header with a really, really long title name
      </h2>
      <div
        className="slds-panel__header-actions"
      >
        <button
          className="slds-button slds-button_icon slds-button_icon-small slds-panel__close"
          title="Collapse Panel Header with a really, really long title name"
        >
          <svg
            aria-hidden={true}
            className="slds-button__icon"
          >
            <use
              xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#close"
            />
          </svg>
          <span
            className="slds-assistive-text"
          >
            Collapse Panel Header with a really, really long title name
          </span>
        </button>
      </div>
    </div>
    <div
      className="slds-panel__body"
    >
      A panel body accepts any layout or component
    </div>
  </div>
</div>
`;

exports[`render Panel component renders Base Panel with secondary actions 1`] = `
<div
  aria-hidden={false}
  className="slds-panel slds-size_medium slds-panel_docked slds-panel_docked-left slds-is-open"
>
  <div
    className="slds-panel__header"
  >
    <h2
      className="slds-panel__header-title slds-text-heading_small slds-truncate"
      title="Panel Header"
    >
      Panel Header
    </h2>
    <div
      className="slds-panel__header-actions"
    >
      <div
        className="slds-dropdown-trigger slds-dropdown-trigger_click slds-is-open"
      >
        <button
          aria-expanded="true"
          aria-haspopup="true"
          className="slds-button slds-button_icon slds-button_icon-small slds-button_icon-border-filled"
          onClick={[Function]}
          title="Show More"
        >
          <svg
            aria-hidden={true}
            className="slds-button__icon"
          >
            <use
              xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#down"
            />
          </svg>
          <span
            className="slds-assistive-text"
          >
            Show More
          </span>
        </button>
        <div
          className="slds-dropdown slds-dropdown_right slds-dropdown_actions"
        >
          <ul
            aria-label="Show More"
            className="slds-dropdown__list"
            role="menu"
          >
            <li
              className="slds-dropdown__item"
              role="presentation"
            >
              <a
                href="#"
                onClick={[Function]}
                role="menuitem"
                tabIndex="0"
              >
                <span
                  className="slds-truncate"
                  title="Action One"
                >
                  Action One
                </span>
              </a>
            </li>
            <li
              className="slds-dropdown__item"
              role="presentation"
            >
              <a
                href="#"
                onClick={[Function]}
                role="menuitem"
                tabIndex="-1"
              >
                <span
                  className="slds-truncate"
                  title="Action Two"
                >
                  Action Two
                </span>
              </a>
            </li>
            <li
              className="slds-dropdown__item"
              role="presentation"
            >
              <a
                href="#"
                onClick={[Function]}
                role="menuitem"
                tabIndex="-1"
              >
                <span
                  className="slds-truncate"
                  title="Action Three"
                >
                  Action Three
                </span>
              </a>
            </li>
          </ul>
        </div>
      </div>
      <button
        className="slds-button slds-button_icon slds-button_icon-small slds-panel__close"
        title="Collapse Panel Header"
      >
        <svg
          aria-hidden={true}
          className="slds-button__icon"
        >
          <use
            xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#close"
          />
        </svg>
        <span
          className="slds-assistive-text"
        >
          Collapse Panel Header
        </span>
      </button>
    </div>
  </div>
  <div
    className="slds-panel__body"
  >
    A panel body accepts any layout or component
  </div>
</div>
`;

exports[`render Panel component renders Base Panel with the deprecated header 1`] = `
<div
  aria-hidden={false}
  className="slds-panel slds-size_medium slds-panel_docked slds-panel_docked-left slds-is-open"
>
  <div
    className="slds-panel__header"
  >
    <h2
      className="slds-panel__header-title slds-text-heading_small slds-truncate"
      title="Deprecated Panel Header"
    >
      Deprecated Panel Header
    </h2>
    <button
      className="slds-button slds-button_icon slds-button_icon-small slds-panel__close"
      title="Collapse Deprecated Panel Header"
    >
      <svg
        aria-hidden={true}
        className="slds-button__icon"
      >
        <use
          xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#close"
        />
      </svg>
      <span
        className="slds-assistive-text"
      >
        Collapse Deprecated Panel Header
      </span>
    </button>
  </div>
  <div
    className="slds-panel__body"
  >
    A panel body accepts any layout or component
  </div>
</div>
`;
